unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs

```diff
@@ -4638,20 +4687,35 @@ public void PlaceHeldObject(string objectId, bool forceAction=false, bool placeS
             InstantiatePrefabTest script = physicsSceneManager.GetComponent<InstantiatePrefabTest>();
             //set degreeIncrement to 90 for placing held objects to check for vertical angles
             List<ReceptacleSpawnPoint> spawnPoints = targetReceptacle.ReturnMySpawnPoints(onlyPointsCloseToAgent);
-            if (randomSeed != 0) {
-                List<ReceptacleSpawnPoint> randomizedSpawnPoints = new List<ReceptacleSpawnPoint>();
-                float maxDistance = z;
-                if (maxDistance == 0.0f) {
+            if (randomSeed != 0 || putNearXY) {
+                List<KeyValuePair<ReceptacleSpawnPoint, float>> distSpawnPoints = new List<KeyValuePair<ReceptacleSpawnPoint, float>>();
+                if (Mathf.Approximately(maxDistance, 0.0f)) {
```

Comment:
Using `Mathf.Approximately` doesn't seem correct here, we're trying to check if the user passed an _exact_ 0 value. That said, I would prefer this was refactored so that `maxDistance` was of type `float?` (so we set it equal to `maxVisibleDistance` if it's null) and we threw and error if anyone passed a value of `maxDistance==0`.

Reply:
this has now been replaced so `maxDistance` uses a nullable type.